ls1intum/Artemis
# Remove repository access from students who were removed from a team

### Checklist

### Motivation and Context
This PR addresses [this](https://github.com/ls1intum/Artemis/issues/1279) issue. So far the `BitbucketService` only provided a method to set the permissions of a student for a repository but there was no way to completely remove all permissions for a repository.

### Description
1. Add new method `removeStudentRepositoryAccess` in `BitbucketService`
2. Use new method when a student gets removed from a team (so that he now loses access completely instead of still being able to read access the repository of his former teams).

### Steps for Testing
**Note:** If you are new to the concept of team-based exercises, have a look at [this](https://github.com/ls1intum/Artemis/pull/1282#issuecomment-605954482) comment first.

1.1. Log in to Artemis as a student who is assigned to a team for a programming exercise
1.2. Navigate to Overview and click on the Course that contains this exercise
1.3. Clone the repository for this team programming exercise
__________________
2.1. In parallel, log in to Artemis as an admin (or instructor / TA in the course)
2.2. Navigate to Course Administration
2.3. Click on Exercises for the course mentioned above
2.4. Click on Teams for the team programming exercise mentioned above
2.5. Remove the student (who was used above) from the team
__________________
1.4. As the student, try `git fetch` for the cloned repository
1.5. Verify that this does not work

Expected error message for the last step:
```
fatal: remote error: Repository not found
The requested repository does not exist, or you do not have permission to
access it.
```
